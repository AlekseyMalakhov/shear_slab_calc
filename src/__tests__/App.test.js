'use strict';
import React from 'react';
import App from '../App.js';
import { create, act } from "react-test-renderer";

test("Простой тест", () => {
    let component;
    act(() => {
        component = create(
          <App></App>
        );
    });
    var app = component.getInstance();
    var mock_state = {
      input_n_load: 50,
      input_a_column_size: 500,
      input_b_column_size: 400,
      input_t_slab_size: 200,
      input_a_slab_size: 30,
      concrete_grade: "b10",
    };
    app.getData(mock_state);
    //console.log(app.state);
    //expect(app.state.report_data.factor).toBe(2.308);
    expect(app.state.report_data).toMatchSnapshot();
   
});

test("Простой тест c моментами", () => {
  let component;
  act(() => {
      component = create(
        <App></App>
      );
  });
  var app = component.getInstance();
  var mock_state = {
    input_n_load: 50,
    input_mx_load: 5,
    input_my_load: 2,
    input_a_column_size: 400,
    input_b_column_size: 200,
    input_t_slab_size: 200,
    input_a_slab_size: 30,
    concrete_grade: "b20",
  };
  app.getData(mock_state);
  //console.log(app.state);
  //expect(app.state.report_data.factor).toBe(2.841);
  expect(app.state.report_data).toMatchSnapshot();
});

test("Армирование", () => {
  let component;
  act(() => {
      component = create(
        <App></App>
      );
  });
  var app = component.getInstance();
  var mock_state = {
    input_n_load: 40,
    input_mx_load: 5,
    input_a_column_size: 500,
    input_b_column_size: 200,
    input_t_slab_size: 200,
    input_a_slab_size: 30,
    concrete_grade: "b30",
    shear_bars_diameter: "8",
    shear_bars_grade: "a240c",
    shear_bars_row_number: 2,
    input_shear_bars_spacing_to_prev: [0, 50, 100],
    shear_reinforcement: true,
    shear_bars_number: {
      X: [0, 5, 7],
      Y: [0, 4, 6],
    },
  };
  app.getData(mock_state);
  //console.log(app.state);
  //expect(app.state.report_data.factor).toBe(0.962);
  expect(app.state.report_data).toMatchSnapshot();
});

test("Колонна рядом с 1 краем плиты без армирования", () => {
  let component;
  act(() => {
      component = create(
        <App></App>
      );
  });
  var app = component.getInstance();
  var mock_state = {
    input_n_load: 50,
    input_mx_load: 5,
    input_my_load: 2,
    input_a_column_size: 400,
    input_b_column_size: 200,
    input_t_slab_size: 200,
    input_a_slab_size: 30,
    concrete_grade: "b20",
    slab_edge: true,
    edge_left: true,
    input_edge_left_dist: 0,
  };

  app.getData(mock_state);
  //console.log(app.state);
  //expect(app.state.report_data.factor).toBe(3.986);
  expect(app.state.report_data).toMatchSnapshot();
});

test("Колонна рядом с 1 краем плиты c армированием", () => {
  let component;
  act(() => {
      component = create(
        <App></App>
      );
  });
  var app = component.getInstance();
  var mock_state = {
    input_n_load: 50,
    input_mx_load: 5,
    input_my_load: 2,
    input_a_column_size: 400,
    input_b_column_size: 200,
    input_t_slab_size: 200,
    input_a_slab_size: 30,
    concrete_grade: "b20",
    slab_edge: true,
    edge_left: true,
    input_edge_left_dist: 0,
    shear_bars_diameter: "8",
    shear_bars_grade: "a240c",
    shear_bars_row_number: 2,
    input_shear_bars_spacing_to_prev: [0, 50, 100],
    shear_reinforcement: true,
    shear_bars_number: {
      X: [0, 5, 7],
      Y: [0, 4, 6],
    },
  };

  app.getData(mock_state);
  //console.log(app.state);
  //expect(app.state.report_data.factor).toBe(2.111);
  expect(app.state.report_data).toMatchSnapshot();
});

test("Колонна на углу плиты без армирования", () => {
  let component;
  act(() => {
      component = create(
        <App></App>
      );
  });
  var app = component.getInstance();
  var mock_state = {
    input_n_load: 50,
    input_mx_load: 5,
    input_my_load: 3,
    input_a_column_size: 400,
    input_b_column_size: 200,
    input_t_slab_size: 200,
    input_a_slab_size: 30,
    concrete_grade: "b20",
    slab_edge: true,
    edge_top: true,
    edge_left: true,
    input_edge_left_dist: 0,
    input_edge_top_dist: 0,
  };

  app.getData(mock_state);
  //console.log(app.state);
  //expect(app.state.report_data.factor).toBe(6.938);
  expect(app.state.report_data).toMatchSnapshot();
});

test("Колонна с отверстиями", () => {
  let component;
  act(() => {
      component = create(
        <App></App>
      );
  });
  var app = component.getInstance();
  var mock_state = {
    input_n_load: 50,
    input_mx_load: 5,
    input_my_load: 3,
    input_a_column_size: 400,
    input_b_column_size: 200,
    input_t_slab_size: 200,
    input_a_slab_size: 30,
    concrete_grade: "b30",
    openingIsNear: true,
    input_openings: {
      X: [0, 500, -300],
      Y: [0, 300, -100],
      a: [0, 400, 100],
      b: [0, 200, 1200],
    },
  };

  app.getData(mock_state);
  //console.log(app.state);
  //expect(app.state.report_data.factor).toBe(4.924);
  expect(app.state.report_data).toMatchSnapshot();
});

test("Колонна с отверстиями и армированием", () => {
  let component;
  act(() => {
      component = create(
        <App></App>
      );
  });
  var app = component.getInstance();
  var mock_state = {
    input_n_load: 50,
    input_mx_load: 5,
    input_my_load: 3,
    input_a_column_size: 400,
    input_b_column_size: 200,
    input_t_slab_size: 200,
    input_a_slab_size: 30,
    concrete_grade: "b30",
    openingIsNear: true,
    input_openings: {
      X: [0, 500, -300],
      Y: [0, 300, -100],
      a: [0, 400, 100],
      b: [0, 200, 1200],
    },
    shear_bars_diameter: "8",
    shear_bars_grade: "a240c",
    shear_bars_row_number: 2,
    input_shear_bars_spacing_to_prev: [0, 50, 100],
    shear_reinforcement: true,
    shear_bars_number: {
      X: [0, 5, 7],
      Y: [0, 4, 6],
    },
  };

  app.getData(mock_state);
  //console.log(app.state);
  //expect(app.state.report_data.factor).toBe(2.843);
  expect(app.state.report_data).toMatchSnapshot();
});