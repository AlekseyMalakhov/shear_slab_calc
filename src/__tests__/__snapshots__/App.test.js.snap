// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Армирование 1`] = `
Object {
  "ab": 353600,
  "asw_sw": 0.87,
  "cut_chars": undefined,
  "cut_off": 0,
  "cut_off_ibx": 0,
  "cut_off_iby": 0,
  "cut_off_sx": 0,
  "cut_off_sy": 0,
  "cut_xc": 0,
  "cut_yc": 0,
  "f_ult": 612.252,
  "factor": 0.962,
  "fb_ult": 365.98,
  "fsw_ult": 246.272,
  "fsw_ult_1": 246.272,
  "h0": 170,
  "ibx": 133173666,
  "iby": 54303666,
  "lx": 670,
  "ly": 370,
  "m_factor_1": 0.419,
  "m_factor_2": 0.321,
  "mbx_ult": 69945.93,
  "mby_ult": 51647.13,
  "mswx_ult": 47067.907,
  "mswx_ult_1": 47067.907,
  "mswy_ult": 34754.307,
  "mswy_ult_1": 34754.307,
  "mx_1": undefined,
  "my_1": undefined,
  "n_factor": 0.641,
  "qsw": 0.148,
  "rbt": 0.001035,
  "size_u_bottom": 670,
  "size_u_left": 370,
  "size_u_right": 370,
  "size_u_top": 670,
  "sx": 0,
  "sy": 0,
  "u": 2080,
  "wbx": 397533,
  "wby": 293533,
  "xa": 0,
  "xc": 0,
  "xmax": 0,
  "xmax_op": 0,
  "ya": 0,
  "yc": 0,
  "ymax": 0,
  "ymax_op": 0,
}
`;

exports[`Колонна на углу плиты без армирования 1`] = `
Object {
  "ab": 130900,
  "asw_sw": 0,
  "cut_chars": undefined,
  "cut_off": 0,
  "cut_off_ibx": 0,
  "cut_off_iby": 0,
  "cut_off_sx": 0,
  "cut_off_sy": 0,
  "cut_xc": 0,
  "cut_yc": 0,
  "f_ult": 106.03,
  "factor": 6.938,
  "fb_ult": 106.03,
  "fsw_ult": 0,
  "fsw_ult_1": 0,
  "h0": 170,
  "ibx": 20063496,
  "iby": 5574329,
  "lx": 485,
  "ly": 285,
  "m_factor_1": 42.224,
  "m_factor_2": 2.313,
  "mbx_ult": 8314.88,
  "mby_ult": 3304.8,
  "mswx_ult": 0,
  "mswx_ult_1": 0,
  "mswy_ult": 0,
  "mswy_ult_1": 0,
  "mx_1": 113888.69,
  "my_1": 94274.69,
  "n_factor": 4.625,
  "qsw": 0,
  "rbt": 0.00081,
  "size_u_bottom": 485,
  "size_u_left": 0,
  "size_u_right": 285,
  "size_u_top": 0,
  "sx": 101837.5,
  "sy": -101837.5,
  "u": 770,
  "wbx": 60384,
  "wby": 24000,
  "xa": 200,
  "xc": 132.26,
  "xmax": 332.26,
  "xmax_op": 0,
  "ya": 100,
  "yc": -132.26,
  "ymax": 232.26,
  "ymax_op": 0,
}
`;

exports[`Колонна рядом с 1 краем плиты c армированием 1`] = `
Object {
  "ab": 227800,
  "asw_sw": 0.9,
  "cut_chars": undefined,
  "cut_off": 0,
  "cut_off_ibx": 0,
  "cut_off_iby": 0,
  "cut_off_sx": 0,
  "cut_off_sy": 0,
  "cut_xc": 0,
  "cut_yc": 0,
  "f_ult": 348.536,
  "factor": 2.111,
  "fb_ult": 184.52,
  "fsw_ult": 164.016,
  "fsw_ult_1": 164.016,
  "h0": 170,
  "ibx": 34764441,
  "iby": 37419333,
  "lx": 485,
  "ly": 370,
  "m_factor_1": 3.888,
  "m_factor_2": 0.704,
  "mbx_ult": 15469.08,
  "mby_ult": 27852.03,
  "mswx_ult": 13750.294,
  "mswx_ult_1": 13750.294,
  "mswy_ult": 24757.358,
  "mswy_ult_1": 24757.358,
  "mx_1": 102708.71,
  "my_1": 19614,
  "n_factor": 1.407,
  "qsw": 0.153,
  "rbt": 0.00081,
  "size_u_bottom": 485,
  "size_u_left": 0,
  "size_u_right": 370,
  "size_u_top": 485,
  "sx": 146675,
  "sy": 0,
  "u": 1340,
  "wbx": 112339,
  "wby": 202266,
  "xa": 200,
  "xc": 109.46,
  "xmax": 309.46,
  "xmax_op": 0,
  "ya": 0,
  "yc": 0,
  "ymax": 0,
  "ymax_op": 0,
}
`;

exports[`Колонна рядом с 1 краем плиты без армирования 1`] = `
Object {
  "ab": 227800,
  "asw_sw": 0,
  "cut_chars": undefined,
  "cut_off": 0,
  "cut_off_ibx": 0,
  "cut_off_iby": 0,
  "cut_off_sx": 0,
  "cut_off_sy": 0,
  "cut_xc": 0,
  "cut_yc": 0,
  "f_ult": 184.52,
  "factor": 3.986,
  "fb_ult": 184.52,
  "fsw_ult": 0,
  "fsw_ult_1": 0,
  "h0": 170,
  "ibx": 34764441,
  "iby": 37419333,
  "lx": 485,
  "ly": 370,
  "m_factor_1": 7.344,
  "m_factor_2": 1.329,
  "mbx_ult": 15469.08,
  "mby_ult": 27852.03,
  "mswx_ult": 0,
  "mswx_ult_1": 0,
  "mswy_ult": 0,
  "mswy_ult_1": 0,
  "mx_1": 102708.71,
  "my_1": 19614,
  "n_factor": 2.657,
  "qsw": 0,
  "rbt": 0.00081,
  "size_u_bottom": 485,
  "size_u_left": 0,
  "size_u_right": 370,
  "size_u_top": 485,
  "sx": 146675,
  "sy": 0,
  "u": 1340,
  "wbx": 112339,
  "wby": 202266,
  "xa": 200,
  "xc": 109.46,
  "xmax": 309.46,
  "xmax_op": 0,
  "ya": 0,
  "yc": 0,
  "ymax": 0,
  "ymax_op": 0,
}
`;

exports[`Колонна с отверстиями 1`] = `
Object {
  "ab": 144330,
  "asw_sw": 0,
  "cut_chars": Object {
    "angles": Array [
      Array [
        36.87,
        74.055,
      ],
      Array [
        199.654,
        333.435,
      ],
    ],
    "cut_ibx": Array [
      6826300,
      8412473,
      30053250,
      7452557,
      7620622,
    ],
    "cut_iby": Array [
      5005406,
      1930420,
      4221083,
      6588312,
      7492879,
    ],
    "cut_midX": Array [
      211.88,
      285,
      -285,
      -188.75,
      -175.54,
    ],
    "cut_midY": Array [
      185,
      133.21,
      0,
      185,
      -185,
    ],
    "cut_sx": Array [
      30987,
      29517,
      -105450,
      -36335,
      -38431,
    ],
    "cut_sy": Array [
      27056,
      13796,
      0,
      35612,
      -40503,
    ],
    "cut_u": Array [
      146.25,
      103.57,
      370,
      192.5,
      218.93,
    ],
    "dir": Array [
      "horiz",
      "vert",
      "vert",
      "horiz",
      "horiz",
    ],
  },
  "cut_off": 1031,
  "cut_off_ibx": 60365202,
  "cut_off_iby": 25238100,
  "cut_off_sx": -119712,
  "cut_off_sy": 35961,
  "cut_xc": 141,
  "cut_yc": -42.36,
  "f_ult": 149.38,
  "factor": 4.924,
  "fb_ult": 149.38,
  "fsw_ult": 0,
  "fsw_ult_1": 0,
  "h0": 170,
  "ibx": 13727829,
  "iby": 20697146,
  "lx": 570,
  "ly": 370,
  "m_factor_1": 23.976,
  "m_factor_2": 1.641,
  "mbx_ult": 5669.81,
  "mby_ult": 16017.08,
  "mswx_ult": 0,
  "mswx_ult_1": 0,
  "mswy_ult": 0,
  "mswy_ult_1": 0,
  "mx_1": 118174.35,
  "my_1": 50192.23,
  "n_factor": 3.283,
  "qsw": 0,
  "rbt": 0.001035,
  "size_u_bottom": 570,
  "size_u_left": 370,
  "size_u_right": 370,
  "size_u_top": 570,
  "sx": 0,
  "sy": 0,
  "u": 849,
  "wbx": 32224,
  "wby": 91032,
  "xa": 0,
  "xc": 0,
  "xmax": 426,
  "xmax_op": 426,
  "ya": 0,
  "yc": 0,
  "ymax": 227.36,
  "ymax_op": 227.36,
}
`;

exports[`Колонна с отверстиями и армированием 1`] = `
Object {
  "ab": 144330,
  "asw_sw": 0.947,
  "cut_chars": Object {
    "angles": Array [
      Array [
        36.87,
        74.055,
      ],
      Array [
        199.654,
        333.435,
      ],
    ],
    "cut_ibx": Array [
      6826300,
      8412473,
      30053250,
      7452557,
      7620622,
    ],
    "cut_iby": Array [
      5005406,
      1930420,
      4221083,
      6588312,
      7492879,
    ],
    "cut_midX": Array [
      211.88,
      285,
      -285,
      -188.75,
      -175.54,
    ],
    "cut_midY": Array [
      185,
      133.21,
      0,
      185,
      -185,
    ],
    "cut_sx": Array [
      30987,
      29517,
      -105450,
      -36335,
      -38431,
    ],
    "cut_sy": Array [
      27056,
      13796,
      0,
      35612,
      -40503,
    ],
    "cut_u": Array [
      146.25,
      103.57,
      370,
      192.5,
      218.93,
    ],
    "dir": Array [
      "horiz",
      "vert",
      "vert",
      "horiz",
      "horiz",
    ],
  },
  "cut_off": 1031,
  "cut_off_ibx": 60365202,
  "cut_off_iby": 25238100,
  "cut_off_sx": -119712,
  "cut_off_sy": 35961,
  "cut_xc": 141,
  "cut_yc": -42.36,
  "f_ult": 258.731,
  "factor": 2.843,
  "fb_ult": 149.38,
  "fsw_ult": 109.351,
  "fsw_ult_1": 109.351,
  "h0": 170,
  "ibx": 13727829,
  "iby": 20697146,
  "lx": 570,
  "ly": 370,
  "m_factor_1": 13.843,
  "m_factor_2": 0.948,
  "mbx_ult": 5669.81,
  "mby_ult": 16017.08,
  "mswx_ult": 4150.451,
  "mswx_ult_1": 4150.451,
  "mswy_ult": 11724.922,
  "mswy_ult_1": 11724.922,
  "mx_1": 118174.35,
  "my_1": 50192.23,
  "n_factor": 1.895,
  "qsw": 0.161,
  "rbt": 0.001035,
  "size_u_bottom": 570,
  "size_u_left": 370,
  "size_u_right": 370,
  "size_u_top": 570,
  "sx": 0,
  "sy": 0,
  "u": 849,
  "wbx": 32224,
  "wby": 91032,
  "xa": 0,
  "xc": 0,
  "xmax": 426,
  "xmax_op": 426,
  "ya": 0,
  "yc": 0,
  "ymax": 227.36,
  "ymax_op": 227.36,
}
`;

exports[`Простой тест 1`] = `
Object {
  "ab": 421600,
  "asw_sw": 0,
  "cut_chars": undefined,
  "cut_off": 0,
  "cut_off_ibx": 0,
  "cut_off_iby": 0,
  "cut_off_sx": 0,
  "cut_off_sy": 0,
  "cut_xc": 0,
  "cut_yc": 0,
  "f_ult": 212.49,
  "factor": 2.308,
  "fb_ult": 212.49,
  "fsw_ult": 0,
  "fsw_ult_1": 0,
  "h0": 170,
  "ibx": 178063666,
  "iby": 139707000,
  "lx": 670,
  "ly": 570,
  "m_factor_1": 0,
  "m_factor_2": 0,
  "mbx_ult": 45541.75,
  "mby_ult": 42000.34,
  "mswx_ult": 0,
  "mswx_ult_1": 0,
  "mswy_ult": 0,
  "mswy_ult_1": 0,
  "mx_1": undefined,
  "my_1": undefined,
  "n_factor": 2.308,
  "qsw": 0,
  "rbt": 0.000504,
  "size_u_bottom": 670,
  "size_u_left": 570,
  "size_u_right": 570,
  "size_u_top": 670,
  "sx": 0,
  "sy": 0,
  "u": 2480,
  "wbx": 531533,
  "wby": 490200,
  "xa": 0,
  "xc": 0,
  "xmax": 0,
  "xmax_op": 0,
  "ya": 0,
  "yc": 0,
  "ymax": 0,
  "ymax_op": 0,
}
`;

exports[`Простой тест c моментами 1`] = `
Object {
  "ab": 319600,
  "asw_sw": 0,
  "cut_chars": undefined,
  "cut_off": 0,
  "cut_off_ibx": 0,
  "cut_off_iby": 0,
  "cut_off_sx": 0,
  "cut_off_sy": 0,
  "cut_xc": 0,
  "cut_yc": 0,
  "f_ult": 258.88,
  "factor": 2.841,
  "fb_ult": 258.88,
  "fsw_ult": 0,
  "fsw_ult_1": 0,
  "h0": 170,
  "ibx": 90972000,
  "iby": 47458666,
  "lx": 570,
  "ly": 370,
  "m_factor_1": 1.671,
  "m_factor_2": 0.947,
  "mbx_ult": 43953.84,
  "mby_ult": 35324.59,
  "mswx_ult": 0,
  "mswx_ult_1": 0,
  "mswy_ult": 0,
  "mswy_ult_1": 0,
  "mx_1": undefined,
  "my_1": undefined,
  "n_factor": 1.894,
  "qsw": 0,
  "rbt": 0.00081,
  "size_u_bottom": 570,
  "size_u_left": 370,
  "size_u_right": 370,
  "size_u_top": 570,
  "sx": 0,
  "sy": 0,
  "u": 1880,
  "wbx": 319200,
  "wby": 256533,
  "xa": 0,
  "xc": 0,
  "xmax": 0,
  "xmax_op": 0,
  "ya": 0,
  "yc": 0,
  "ymax": 0,
  "ymax_op": 0,
}
`;
